### ISSUE TITLE
terminal disappear when change font size from settings

### ISSUE THREAD

**Issue by hansanghoon** [Microsoft/vscode#58159]
Issue Type: <b>Bug</b>

I use oh-my-zsh and set integrated terminal /bin/zsh

1. type cmd+, to show settings tab
2. open terminal(ctrl+shift+`)
3. user settings > terminal > integrated: Font size option. say we have default value 12. 
4. backspace. now value is 1 and terminal gets so small
5. type 1 so now value is 11. and terminal is gone. expected to see current terminal cursor.
6. if i type lots of enters or ctrl+c, then I can see my terminal again.
7. after i saw my terminal again, it looks like change font size 11 -> 1 -> 12 doesn't make terminal to disappear.

VS Code version: Code 1.27.1 (5944e81f3c46a3938a82c701f96d7a59b074cfdc, 2018-09-06T09:14:32.385Z)
OS version: Darwin x64 17.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz (8 x 2500)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.02GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron -psn_0_540804|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (7)</summary>

Extension|Author (truncated)|Version
---|---|---
gitlens|eam|8.5.6
code-runner|for|0.9.4
beautify|Hoo|1.4.2
Go|ms-|0.6.89
vetur|oct|0.12.6
vscode-docker|Pet|0.2.0
java|red|0.30.0

</details>

**Comment by hansanghoon** [comment]
also vertical scroll of terminal get somewhat screwed even after I see my terminal again by typing enter or ctrl+c

**Comment by Git-Lior** [comment]
this seems to be related to the usage of `charMeasure` from xterm:

https://github.com/Microsoft/vscode/blob/4e4e6f9b44dd9e6a53d1f08565c52b8ea9f61a5e/src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts#L140-L149

a similar problem with an explanation of a solution can be found here:

https://github.com/Microsoft/vscode/blob/b54b779c0c896b450561dd2847d27417c104f7c3/src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts#L673-L676

In short, after the fontSize changes, the terminal renders before the xterm `charMeasure` field is updated, so the render uses previous values (corresponding to the previous font size values)

Related: #54678, xtermjs/xterm.js#702

**Comment by roblourens** [comment]
If `1` is an invalid font size for that setting, can the terminal ignore it or set a minimum in the setting schema?

**Comment by Tyriar** [comment]
@roblourens after thinking about this some more I think this is totally on my end as something weird happens when changing the setting frequently. There is already a minimum value which is why VS Code doesn't crash 😄
